backend/celery_worker.py

```diff
@@ -180,38 +179,113 @@ def ping_telemetry():
+            unique_customers[customer["email"]] = customer
+    customers = list(unique_customers.values())
+    logger.debug(f"Unique Customers with latest created date: {customers}")
+
+    # Matching Products
+    matching_product_settings = (
+        supabase_db.table("product_to_features").select("*").execute()
+    ).data
+    logger.debug(f"Matching product settings: {matching_product_settings}")
+
+    # if customer.id in subscriptions.customer then find the user id in the table users where email = customer.email and then update the user_settings with is_premium = True else delete the user_settings
+
+    for customer in customers:
+        logger.debug(f"Customer: {customer}")
+        # Find the subscription of the customer
+        user_id = None
+        matching_subscription = [
+            subscription
+            for subscription in subscriptions
+            if subscription["customer"] == customer["id"]
+        ]
+        logger.debug(f"Matching subscription: {matching_subscription}")
+        user_id = (
+            supabase_db.table("users")
+            .select("id")
+            .filter("email", "eq", customer["email"])
+            .execute()
+        ).data
+        if len(user_id) > 0:
+            user_id = user_id[0]["id"]
+        else:
+            logger.debug(f"User not found for customer: {customer}")
+            continue
+        if len(matching_subscription) > 0:
+
+            # Get the matching product from the subscription
+            matching_product_settings = [
+                product
+                for product in matching_product_settings
+                if product["stripe_product_id"]
+                == matching_subscription[0]["attrs"]["items"]["data"][0]["plan"][
+                    "product"
+                ]
+            ]
+            # Update the user with the product settings
+            supabase_db.table("user_settings").update(
+                {
+                    "max_brains": matching_product_settings[0]["max_brains"],
+                    "max_brain_size": matching_product_settings[0]["max_brain_size"],
+                    "monthly_chat_credit": matching_product_settings[0][
+                        "monthly_chat_credit"
+                    ],
+                    "api_access": matching_product_settings[0]["api_access"],
+                    "models": matching_product_settings[0]["models"],
+                    "is_premium": True,
+                }
+            ).match({"user_id": str(user_id)}).execute()
+        else:
+            # check if user_settings is_premium is true then delete the user_settings
+            user_settings = (
+                supabase_db.table("user_settings")
+                .select("*")
+                .filter("user_id", "eq", user_id)
+                .filter("is_premium", "eq", True)
+                .execute()
+            ).data
+            if len(user_settings) > 0:
+                supabase_db.table("user_settings").delete().match(
```

Comment:
Small question here if `len(match_subscription) == 0` do we delete the row or update the `is_premium` to False ?

Reply:
I'm deleting the row because then when the user ask a new question it will insert with the models, questions and all by default. Otherwise I have to look for the default value of the db manually